plutus-ledger/src/Ledger/Tx/CardanoAPI.hs

```diff
@@ -127,6 +127,16 @@ toCardanoScriptWitness datum redeemer (P.Versioned script lang) = (case lang of
   ) <*> pure datum
     <*> pure (C.fromPlutusData $ PV1.toData redeemer)
     <*> pure zeroExecutionUnits
+toCardanoScriptWitness datum redeemer (Right (P.Versioned ref lang)) = (case lang of
```

Comment:
It should be possible to merge this code with the code for `toCardanoTxInWitness`.